sampleExtension/background.js

```diff
@@ -1,60 +1,117 @@
+        interceptCounter = counter;
+        return callback();
     });
 };
 
 // This function adds one url to the blacklist
 addToBlockedSites = function(urlToAdd) {
-    var formatUrl = urlToAdd.split(['//'])[1];
-    formatUrl = '*://' + formatUrl + '*';
-    blockedSites.push(formatUrl);
-    chrome.storage.sync.set({"tds_blacklist" : blockedSites }, function() {
-        if(chrome.runtime.error) {
-            console.log("Runtime error.");
-        }
-    });
+    var blockedSiteItem = new BlockedSite(formatUrl(urlToAdd));
+    blockedSites.push(blockedSiteItem);
+    updateStorage();
+    replaceListener();
+};
+
+// This function adds the current time+date to the saved time+date list
+addToInterceptDateList = function() {
+    var newDate = new Date().toDateString();
+    if (interceptDateList == null) {
+        interceptDateList = [newDate];
+    } else {
+        interceptDateList.push(newDate);
+    }
+    setInterceptDateList(interceptDateList);
+};
+
+//TODO fix in interation 3 for url-handling module
+formatUrl = function(url) {
+    result = url.split(['//'])[1];
+    result = result.split("").reverse().join("");
+    result = result.split(['/'])[1];
+    result = result.split("").reverse().join("");
+    return result;
 };
 
 /* --------------- ------ Listener functions ------ ---------------*/
 
 replaceListener = function() {
-    chrome.webRequest.onBeforeRequest.removeListener(intercept);
-    addWebRequestListener();
+    chrome.webRequest.onBeforeRequest.removeListener(handleInterception);
+    urlList = filterBlockedSitesOnChecked();
+    if (urlList.length > 0) {
+        addWebRequestListener(urlList);
+    }
+};
+
+filterBlockedSitesOnChecked = function() {
+    if(blockedSites.length > 0) {
+        return blockedSites.filter(function (a) {return a.checkboxVal == true;});
+    }
+    return [];
 };
 
-addWebRequestListener = function() {
-    if(blockedSites != null && blockedSites.length > 0) {
-        chrome.webRequest.onBeforeRequest.addListener(
-                intercept,
-                {
-                    //Url's to be intercepted
-                    urls: blockedSites,
-                    types: ["main_frame"]
-            },
-            ["blocking"]
-        );
+addWebRequestListener = function(urlList) {
+    chrome.webRequest.onBeforeRequest.addListener(
+        handleInterception
+        ,{
+            urls: urlList.map(function (a) {return a.url;})
+            ,types: ["main_frame"]
+        }
+        ,["blocking"]
+    );
+};
+
+intercept = function(details) {
+    incrementInterceptionCounter(details.url);
+    addToInterceptDateList();
+    seStorageOriginalDestination(details.url);
+    setEnableInterceptionAfterTimeout(false, 2000);
+    return {redirectUrl: redirectLink};
+};
+
+handleInterception = function(details) {
+    if (interceptionEnabled) {
+        return intercept(details);
+    } else {
+        setEnableInterceptionAfterTimeout(true, 1000);
```

Comment:
It's a hard-coded delay, what if I have any other number in there? Where does the decision to have 1000 as the delay come from? Trial and error? This type of things introduces inconsistencies in the inner workings of your program.